File: back/internal/infra/dto/todo.go
package dto

import "time"

type Todo struct {
	ID        uint      `json:"id" gorm:"primaryKey"`
	Title     string    `json:"title" gorm:"size:512;not null"`
	Completed bool      `json:"completed" gorm:"default:false;not null"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	Files     []File    `json:"files,omitempty" gorm:"many2many:todos_files"`
}

File: front/lib/types/todo.ts
/** Go `dto.Todo` の JSON（`files` は `todos_files` 経由で結合された `dto.File`） */
export type TodoFile = {
  id: number;
  name: string;
  created_at: string;
  updated_at: string;
};

export type Todo = {
  id: number;
  title: string;
  completed: boolean;
  created_at: string;
  updated_at: string;
  files?: TodoFile[];
};
